File: Plugins/Orimath.Basics.View/ViewModels/InstructionLineViewModel.cs
using Mvvm;
using Orimath.Plugins;
using Orimath.FoldingInstruction;

namespace Orimath.Basics.View.ViewModels
{
    public class InstructionLineViewModel : NotifyPropertyChanged
    {
        private readonly IViewPointConverter _pointConverter;

        private double _x1;
        public double X1 { get => _x1; set => SetValue(ref _x1, value); }

        private double _x2;
        public double X2 { get => _x2; set => SetValue(ref _x2, value); }

        private double _y1;
        public double Y1 { get => _y1; set => SetValue(ref _y1, value); }

        private double _y2;
        public double Y2 { get => _y2; set => SetValue(ref _y2, value); }

        private InstructionColor _color;
        public InstructionColor Color { get => _color; set => SetValue(ref _color, value); }    

        public InstructionLineViewModel(IViewPointConverter pointConverter)
        {
            _pointConverter = pointConverter;
        }

        public InstructionLineViewModel(IViewPointConverter pointConverter, InstructionLine model)
            : this(pointConverter)
        {
            SetModel(model);
        }

        public void SetModel(InstructionLine model)
        {
            var point1 = _pointConverter.ModelToView(model.Line.Point1);
            var point2 = _pointConverter.ModelToView(model.Line.Point2);
            X1 = point1.X;
            Y1 = point1.Y;
            X2 = point2.X;
            Y2 = point2.Y;
            Color = model.Color;
        }
    }
}

File: Plugins/Orimath.Basics.View/ViewModels/InstructionLineViewModel.fs
namespace Orimath.Basics.View.ViewModels
open Orimath.Controls
open Orimath.Plugins
open Orimath.Combination
open ApplicativeProperty

type InstructionLineViewModel(pointConverter: IViewPointConverter) =
    inherit NotifyPropertyChanged()
    member val X1 = Prop.value 0.0
    member val X2 = Prop.value 0.0
    member val Y1 = Prop.value 0.0
    member val Y2 = Prop.value 0.0
    member val Color = Prop.value InstructionColor.Black

    member this.SetModel(model: InstructionLine) =
        let p1 = pointConverter.ModelToView(model.Line.Point1)
        let p2 = pointConverter.ModelToView(model.Line.Point2)
        this.X1.Value <- p1.X
        this.Y1.Value <- p1.Y
        this.X2.Value <- p2.X
        this.Y2.Value <- p2.Y
        this.Color.Value <- model.Color

    static member Create(pointConverter, model) =
        let vm = InstructionLineViewModel(pointConverter)
        vm.SetModel(model)
        vm
